tests/HeaderFactoryTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders\Tests;
+
+use Aidantwoods\SecureHeaders\HeaderFactory;
+use PHPUnit_Framework_TestCase;
+
+class HeaderFactoryTest extends PHPUnit_Framework_TestCase
+{
+    public function testCSPHeaders()
+    {
+        $CSPHeader = HeaderFactory::build(
+            'Content-SECURITY-Policy',
+            "default-src 'none'"
+        );
+
+        $CSPROHeader = HeaderFactory::build(
+            'ConTeNt-Security-Policy-Report-Only',
+            "default-src 'none'"
+        );
+
+        $XCSPHeader = HeaderFactory::build(
+            'x-Content-Security-POLICY',
+            "default-src 'none'"
+        );
+
+        $XCSPROHeader = HeaderFactory::build(
+            'X-content-security-policy-report-only',
+            "default-src 'none'"
+        );
+
+        $Headers = [$CSPHeader, $CSPROHeader, $XCSPHeader, $XCSPROHeader];
+
+        foreach ($Headers as $Header)
+        {
+            $this->assertSame(
+                'Aidantwoods\SecureHeaders\Headers\CSPHeader',
```

Comment:
Thinking about that now, not sure if it would make sense to refactor `RegularHeader`'s contents into an abstract base so we can have method sharing while having separate instanceable types.